pkg/provider/aws/secretsmanager/secretsmanager.go

```diff
@@ -380,7 +380,50 @@ func (sm *SecretsManager) findByTags(ctx context.Context, ref esv1.ExternalSecre
 		})
 	}
 
-	return sm.fetchWithBatch(ctx, filters, nil)
+	secretListIds, err := sm.filterTagByKeyValPair(ctx, filters, ref)
+	if err != nil {
+		return nil, fmt.Errorf("failed to filter tags correctly: %w", err)
+	}
+
+	log.V(5).Info("secretListIds", "secrets", secretListIds)
+	return sm.fetchWithBatch(ctx, fetchWithBatchOpts{secretsListIds: secretListIds})
+}
+
+func (sm *SecretsManager) filterTagByKeyValPair(ctx context.Context, filters []types.Filter, ref esv1.ExternalSecretFind) ([]string, error) {
+	listSecrets, err := sm.client.ListSecrets(ctx, &awssm.ListSecretsInput{
+		Filters: filters,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to list secrets: %w", err)
+	}
```

Comment:
This entire thing changes how Batch works now. It first Lists secrets which adds API calls to the entire thing which is not good. The entire purpose of Batch fetching was that it prevents having to list or get all secrets. It does that in one call. Now this is a regression since there is always a ListSecrets call. This needs to be done somehow differently.

Reply:
That's a fair point. I'm basically trying to workaround a limitation (the tag filters) in the AWS SecretsManager API, by performing an additional API call to filter down the list of secrets. 

If [BatchGetSecretValue](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/secretsmanager/batch-get-secret-value.html) returned the tags with the secrets I could filter them down without an additional API call.
Unfortunately I can't think of another way to implement this feature without the extra API call overhead.

With that said, I'll give it some thought, and if I can't come up with a better solution I'll close the PR (and probably switch to `ParameterStore` since it supports searching by key value pair). 

Thanks for the review.

Reply:
This is really unfortunate. I see how the prefix will include other secrets. And since batch isn't returning the Tags for whatever f*cking reason, there is little that we can do. Hmmmmm. Maybe it's fine. It's just ONE extra call. After that I see that you basically batch fetch with the IDs.

That is really unfortunate. I hate it. :D But you're right, I don't know of a better way. We can't post process.

Reply:
Please fix the pagination then, and after that it should be okay.